src/IdentityProviders/IdentityProvider.js

```diff
@@ -0,0 +1,7 @@
+
+export default class IdentityProvider {
+    generatePaylod(){}
```

Comment:
Following convention used by the project you could implement the following for each method:
```
throw new Error('Must be implemented by a sub-class');
```